CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
-project(prime_server LANGUAGES CXX C VERSION 0.6.7)
+project(prime_server LANGUAGES CXX C VERSION 0)
```

Comment:
i assume that the `set(VERSION ${PROJECT_VERSION})` below had the same effect as this. if thats the case i dont see anything wrong with this.

one other option is to move the `Get version` snippet below into a function in a `cmake` directory and just call it at the beginning of all this and still set the version in the `project` call

Reply:
That is a weird part which I dont fully understand because when `VERSION` is removed from that line `set_property` fails later even though it should not be related but tbh I didn't do much debugging.

Reply:
Do note that if `VERSION` is not defined outside of `project` call pkgconf file will not have it properly substituted.